src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -1122,6 +1122,28 @@ public override void VisitDeclarationPattern(IDeclarationPatternOperation operat
             Assert.Empty(operation.Children);
         }
 
+        public override void VisitRecursivePattern(IRecursivePatternOperation operation)
+        {
+            Assert.Equal(OperationKind.RecursivePattern, operation.Kind);
+
+            if (!((operation.Syntax as CSharp.Syntax.RecursivePatternSyntax)?.Designation).IsKind(CSharp.SyntaxKind.DiscardDesignation))
+            {
+                Assert.NotNull(operation.DeclaredSymbol);
+            }
+
+            IEnumerable<IOperation> children = Enumerable.Empty<IOperation>();
+            if (!operation.DeconstructionSubpatterns.IsDefault)
+            {
+                children = children.Concat(operation.DeconstructionSubpatterns);
+            }
+            if (!operation.PropertySubpatterns.IsDefault)
+            {
+                children = children.Concat(operation.PropertySubpatterns.Select(x => x.Item2));
+            }
+
+            AssertEx.Equal(children, operation.Children);
+        }
```

Comment:
>} [](start = 8, length = 1)

Consider verifying that the symbol part of PropertySubpatterns contains non-null symbols.

Reply:
Noted in #32012 for follow-up.

---
In reply to: [244034176](https://github.com/dotnet/roslyn/pull/31967#discussion_r244034176) [](ancestors = 244034176)